kubernetes-client/src/main/java/io/fabric8/kubernetes/client/ClientState.java

```diff
@@ -16,8 +16,6 @@
 
 package io.fabric8.kubernetes.client;
 
-@Deprecated
-public interface ClientAware<C extends Client> {
-
-  C getClient();
+public interface ClientState extends HttpClientAware, ConfigAware<Config> {
```

Comment:
Not convinced by the `ClientState` name, which seems to imply it provides a state rather than the fact that it holds the client and its config… Maybe `ClientStateHolder` (or something similar) would be better?

Reply:
@rohanKanojia @manusa what name would you like to see for ClientState?

Reply:
Hmm, I'm not good with naming things :roll_eyes: . Does `ClientContext` make sense?

Reply:
@manusa we have:

- ClientState
- ClientStateHolder
- ClientContext

If you support any one of these, let's use it.  Or do you want something else?

Reply:
Sorry for the delayed reply. I really have no preference about the name for this.

Is this something that users are going to actually implement? I don't see it very likely.

I see that OperationContext extends this interface, so maybe ClientContext might be a reasonable choice.

Reply:
> Is this something that users are going to actually implement? I don't see it very likely.

Ideally no.

> I see that OperationContext extends this interface, so maybe ClientContext might be a reasonable choice.

OperationContext, OperationSupport, and Client all extend/implement this interface.  OperationSupport just unwraps/holds an OperationContext, so for clarity we could not extend there and just pass the OperationContext as needed.

I'll update to this new name.